sky/clouds/kubernetes.py

```diff
@@ -197,8 +232,12 @@ def regions_with_offering(cls, instance_type: Optional[str],
             # provision_timeout, after which failover will be triggered.
             for r in regions:
                 context = r.name
-                fits, reason = kubernetes_utils.check_instance_fits(
-                    context, instance_type)
+                try:
+                    fits, reason = kubernetes_utils.check_instance_fits(
+                        context, instance_type)
+                except exceptions.ResourcesUnavailableError as e:
+                    cls.mark_context_unavailable(context, str(e))
+                    continue
```

Comment:
Is it ok to add the context to unavailable list? Would it be possible that the resources requirement changes afterward during the failover (e.g., [multiple candidate resources](https://docs.skypilot.co/en/latest/examples/auto-failover.html#multiple-candidate-resources) are specified) and the context should be retried even though the previous resource requirement does not fit in the kubernetes cluster?

Reply:
Discussed with @Michaelvll offline, minutes:

- `sky serve` and `sky jobs` need to failover between regions and clouds in endless loop, so:
  - permanent error should be skipped and the entire region/cloud can be blocked;
  - transient error (literally every error but permanent, since `sky` cannot tell whether an error is really transient) should not break failover and preferred to be retried later;
- It is okay to retry and skip stale kubernetes context for multiple times since this case would be rare. Also, users can easily aware such issues and get rid of the retry overhead by modifying `allowed_context` or run `sky check`;